services/vmagent-config-update/models/config.go

```diff
@@ -0,0 +1,122 @@
+package models
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"gopkg.in/yaml.v2"
+)
+
+type ConfigOptions func(config *Config)
+
+func NewConfig(options ...ConfigOptions) *Config {
+	c := &Config{}
+	for _, fn := range options {
+		fn(c)
+	}
+	return c
+}
+
+// Config represents essential parts from Prometheus config defined at https://prometheus.io/docs/prometheus/latest/configuration/configuration/
+type Config struct {
+	Global                   GlobalConfig   `yaml:"global"`
+	ScrapeConfigs            []ScrapeConfig `yaml:"scrape_configs,omitempty"`
+	targetCount              int
+	targetName               string
+	percentOfUpdatingTargets int
+	stConfigs                []StaticConfig
+}
+
+// GlobalConfig represents essential parts for `global` section of Prometheus config.
+//
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/
+type GlobalConfig struct {
+	ScrapeInterval time.Duration `yaml:"scrape_interval,omitempty"`
+}
+
+// ScrapeConfig represents essential parts for `scrape_config` section of Prometheus config.
+//
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config
+type ScrapeConfig struct {
+	JobName        string         `yaml:"job_name"`
+	ScrapeInterval time.Duration  `yaml:"scrape_interval,omitempty"`
+	ScrapeTimeout  time.Duration  `yaml:"scrape_timeout,omitempty"`
+	StaticConfigs  []StaticConfig `yaml:"static_configs,omitempty"`
+}
+
+// StaticConfig represents essential parts for `static_config` section of Prometheus config.
+//
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#static_config
+type StaticConfig struct {
+	Targets []string          `yaml:"targets"`
+	Labels  map[string]string `yaml:"labels,omitempty"`
+}
+
+func WithGlobalConfig(scrapeInterval time.Duration) ConfigOptions {
+	return func(config *Config) {
+		config.Global = GlobalConfig{ScrapeInterval: scrapeInterval}
+	}
+}
+
+func WithScrapeConfig(targetCount, percentOfUpdatingTargets int, targetName string) ConfigOptions {
+	return func(config *Config) {
+		config.targetCount = targetCount
+		config.targetName = targetName
+		config.percentOfUpdatingTargets = percentOfUpdatingTargets
+		config.update()
+	}
+}
+
+func (cfg *Config) update() {
+	var scrapeConfigs []ScrapeConfig
+	configs := make(map[string][]StaticConfig)
+	configs["vmalert"] = []StaticConfig{
+		{
+			Targets: []string{"localhost:8429"},
+			Labels:  nil,
+		},
+	}
+	if len(cfg.stConfigs) == cfg.targetCount {
+		num := float64(cfg.targetCount) * (float64(cfg.percentOfUpdatingTargets) / 100)
+		if num < 0 {
+			num = 1
+		}
+		for i := 0; i < int(num); i++ {
```

Comment:
I think it might make sense to shuffle the list before doing this. Otherwise, only first targets in the list will be updated, while the rest will always remain constant.

Reply:
added